sklearn/ensemble/_forest.py

```diff
@@ -615,24 +735,29 @@ def _get_oob_predictions(tree, X):
             y_pred = np.rollaxis(y_pred, axis=0, start=3)
         return y_pred
 
-    def _set_oob_score_and_attributes(self, X, y):
+    def _set_oob_score_and_attributes(self, X, y, sample_weight):
         """Compute and set the OOB score and attributes.
 
         Parameters
         ----------
         X : array-like of shape (n_samples, n_features)
             The data matrix.
+
         y : ndarray of shape (n_samples, n_outputs)
             The target matrix.
+
+        sample_weight : ndarray of shape (n_samples,)
+            Sample weights.
         """
-        self.oob_decision_function_ = super()._compute_oob_predictions(X, y)
-        if self.oob_decision_function_.shape[-1] == 1:
+        oob_predictions = super()._compute_oob_predictions(X)
+        if oob_predictions.shape[-1] == 1:
             # drop the n_outputs axis if there is a single output
-            self.oob_decision_function_ = self.oob_decision_function_.squeeze(
-                axis=-1
-            )
+            oob_predictions = oob_predictions.squeeze(axis=-1)
+
+        self.oob_decision_function_ = oob_predictions
         self.oob_score_ = accuracy_score(
-            y, np.argmax(self.oob_decision_function_, axis=1)
+            y, np.argmax(self.oob_decision_function_, axis=1),
+            sample_weight=sample_weight
```

Comment:
Here, this is a bit straightforward addition regarding `sample_weight` but I don't recall that we have a test for that.

It could be good to add that having the correspondence between `sample_weight=None` and `sample_weight=np.ones(...)`.

It would also be nice to come up with a test where the weights are non-unit ones. It might be a bit tricker but I don't have a good example right now.